packs/spell-effects/spell-effect-corrosive-body.json

```diff
@@ -44,9 +44,42 @@
                 "diceNumber": 1,
                 "key": "DamageDice",
                 "predicate": [
-                    "acid"
+                    "acid",
+                    {
+                        "not": "item:corrosive-body"
+                    }
                 ],
                 "selector": "spell-damage"
+            },
+            {
+                "adjustName": true,
+                "choices": [
+                    {
+                        "label": "PF2E.SpellLevel7",
+                        "predicate": [
+                            {
+                                "lte": [
+                                    "parent:level",
+                                    8
+                                ]
+                            }
+                        ],
+                        "value": {
+                            "unarmedDice": 1
+                        }
+                    },
+                    {
+                        "label": "PF2E.SpellLevel9",
+                        "predicate": [
+                            "parent:level:9"
+                        ],
+                        "value": {
+                            "unarmedDice": 2
+                        }
+                    }
+                ],
+                "flag": "corrosiveBody",
+                "key": "ChoiceSet"
             }
```

Comment:
To explain, we don't need the choice sets here and this would fail for a 10th level casting of the spell anyways. This is more compactly done with a ternary in the dice number field instead.

Reply:
Woof completely forgot about 10th level spells. I'll fix with the ternary.